runtime/protocol/http-client-engines/http-client-engine-default/common/src/aws/smithy/kotlin/runtime/http/engine/HttpEngineConfigImpl.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.engine
+
+import aws.smithy.kotlin.runtime.ClientException
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.http.config.EngineFactory
+import aws.smithy.kotlin.runtime.http.config.HttpEngineConfig
+import aws.smithy.kotlin.runtime.http.engine.internal.manageIfPossible
+
+private typealias ConfigApplicator = HttpClientEngineConfig.Builder.() -> Unit
+
+@InternalApi
+public class HttpEngineConfigImpl private constructor(override val httpClient: HttpClientEngine) : HttpEngineConfig {
```

Comment:
missing tests for this/builder and various supplier states + exceptions?

Reply:
Yes, I'll add some.